risc0/circuit/rv32im/src/prove/emu/testutil.rs

```diff
@@ -36,19 +34,36 @@ impl Syscall for NullSyscall {
     }
 }
 
-pub fn basic() -> Program {
+/// Constructs a program from an iterator of instructions starting from an entrypoint.
+fn program_from_instructions(entry: u32, instructions: impl IntoIterator<Item = u32>) -> Program {
+    let mut pc = entry;
+
     Program {
-        entry: 0x4000,
-        image: BTreeMap::from([
-            (0x4000, 0x1234b137), // lui x2, 0x1234b000
-            (0x4004, 0xf387e1b7), // lui x3, 0xf387e000
-            (0x4008, 0x003100b3), // add x1, x2, x3
-            (0x400c, 0x000045b7), // lui a1, 0x4
-            (0x4010, 0x00000073), // ecall(halt)
-        ]),
```

Comment:
It's actually more convenient to have it in this form since sometimes I need to check that jump offsets are correct and when I'm looking at a specific instruction during debugging it's good to know the PC.